ckan-2.10/base/Dockerfile.py-3.10

```
@@ -0,0 +1,92 @@
+# use the official Python 3.10 (bookworm slim) image as the base image
+FROM python:3.10-slim-bookworm
+
+# Tag passed through via the Makefile
+ARG CKAN_TAG=${CKAN_TAG}
+
+ENV TZ=UTC
+ENV APP_DIR=/srv/app
+ENV SRC_DIR=${APP_DIR}src
+ENV CKAN_INI=${APP_DIR}/ckan.ini
+ENV PIP_SRC=${SRC_DIR}
+ENV CKAN_STORAGE_PATH=/var/lib/ckan
+ENV GIT_URL=https://github.com/ckan/ckan.git
+
+# Customize these in the environment (.env) file if needed
+ENV CKAN_SITE_URL=http://localhost:5000
+ENV CKAN__PLUGINS image_view text_view recline_view datastore envvars
+
+# UWSGI options
+ENV UWSGI_HARAKIRI=50
+
+WORKDIR ${APP_DIR}
+
+# Set up timezone
+RUN echo ${TZ} > /etc/timezone
+
+# Set LC_ALL=en_US.UTF-8 will ensure that all locale-dependent operations in the current environment 
+# will use English language and United States cultural conventions with UTF-8 character encoding
+ENV LC_ALL=en_US.UTF-8
+
+# Set the locale          
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y locales
+RUN sed -i "/$LC_ALL/s/^# //g" /etc/locale.gen
+RUN dpkg-reconfigure --frontend=noninteractive locales 
+RUN update-locale LANG=${LC_ALL}
+
+# Install system libraries
+RUN apt-get install --no-install-recommends -y \
+        apt-utils \
+        git \
+        libpq-dev \
+        g++ \
+        linux-headers-generic \
+        libtool \
+        wget
+
+# Create the src directory
+RUN mkdir -p ${SRC_DIR}
+
+# Install supervisord and create the supervisord.d directory
+RUN apt-get install --no-install-recommends -y \
+        supervisor && \
+        mkdir /etc/supervisord.d
+
+COPY setup/supervisord.conf /etc
+
+# Install uwsgi, the CKAN application, the dependency packages for CKAN plus some confiquration
+RUN python3 -m venv ${APP_DIR} && \
+    . ${APP_DIR}/bin/activate && \
+    pip3 install uwsgi && \
```

It's always nice to get the latest pip version

```suggestion
    curl -o ${SRC_DIR}/get-pip.py https://bootstrap.pypa.io/get-pip.py && \
    python3 ${SRC_DIR}/get-pip.py && \
    pip3 install uwsgi && \
```

---

We're pulling this from the python base images?  We shouldn't need to update pip here.

---

Pulling the latest `python:3.10-slim-bookworm` gives me pip 23.x
```
docker pull python:3.10-slim-bookworm
docker run python:3.10-slim-bookworm pip --version
pip 23.0.1 from /usr/local/lib/python3.10/site-packages/pip (python 3.10)
```

Adding the two lines gets me pip 24.0 in the ckan images (in the venv pip).

But on second thought you are right that pip will be there, so we can just do:
```suggestion
    pip3 install -U pip
    pip3 install uwsgi && \
```